src/pages/content-edit/head.js

```diff
@@ -69,32 +62,26 @@ export function controller(options) {
         var start, end;
 
         if(ctrl.schedule && ctrl.start.date) {
-            start = moment(
-                ctrl.start.date + " " + (ctrl.start.time || "00:00"),
-                "YYYY-MM-DD HH:mm"
-            );
+            start = ctrl.start.date + " " + (ctrl.start.time || "00:00");
         } else {
-            start = moment().subtract(10, "seconds");
+            start = subSeconds(Date.now(), 10);
         }
 
         if(ctrl.schedule && ctrl.end.date) {
-            end = moment(
-                ctrl.end.date + " " + (ctrl.end.time || "00:00"),
-                "YYYY-MM-DD HH:mm"
-            );
+            end = ctrl.end.date + " " + (ctrl.end.time || "00:00");
 
-            if(end.isSameOrBefore(start)) {
+            if(compareDesc(end, start) !== -1) {
                 // TODO: handle
                 return console.error("Invalid end date");
             }
         }
 
         return ref.update({
             // TODO: Remove `published` field, it's deprecated
-            published      : start.valueOf(),
-            published_at   : start.valueOf(),
+            published      : parseInt(format(start, "x"), 10),
+            published_at   : parseInt(format(start, "x"), 10),
```

Comment:
Yeah - couldn't find a `format` option that output it straight to an integer. I could cache the parsed value of `start` in this instance but didn't worry about it since `published` is marked for removal everywhere.